clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryUtils.java

```diff
@@ -203,17 +201,13 @@ public static byte[] compress(byte[] raw, CompressionType compressionType) throw
 
     public static ByteBuffer decompress(byte[] metrics, CompressionType compressionType) {
         ByteBuffer data = ByteBuffer.wrap(metrics);
-        try (InputStream in = compressionType.wrapForInput(data, RecordBatch.CURRENT_MAGIC_VALUE, BufferSupplier.create());
-            ByteArrayOutputStream out = new ByteArrayOutputStream()) {
-
+        try (InputStream in = compressionType.wrapForInput(data, RecordBatch.CURRENT_MAGIC_VALUE, BufferSupplier.create())) {
             byte[] bytes = new byte[data.capacity() * 2];
-            int nRead;
-            while ((nRead = in.read(bytes, 0, bytes.length)) != -1) {
+            int nRead = in.read(bytes, 0, bytes.length);
```

Comment:
we can't remove the while loop since the `InputStream#read` does NOT guarantee to read ALL data at once. Please take a look at API spec: https://docs.oracle.com/javase/8/docs/api/java/io/InputStream.html#read-byte:A-int-int-

Reply:
Thanks for the reminder. Since the initialization of ByteBufferOutputStream requires the capacity. I use totalReads to record the total read bytes.